protocol/messenger_handler.go

```diff
@@ -1603,7 +1643,7 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 		state.Response.RequestsToJoinCommunity = append(state.Response.RequestsToJoinCommunity, updatedRequest)
 	}
 
-	if requestToJoinResponseProto.Accepted {
+	if requestToJoinResponseProto.Accepted && myCancelledRequestToJoin.State != communities.RequestToJoinStateCanceled {
```

Comment:
Not sure I understand why we need `myCancelledRequestToJoin.State != communities.RequestToJoinStateCanceled`. Isn't `MyCanceledRequestToJoinForCommunityID` returning only requests in cancelled state?

Reply:
If the user has cancelled his request and we have that value locally, Even if admin auto-approved him we shouldn't make him a member.

Reply:
Can `myCancelledRequestToJoin` be in different state than `RequestToJoinStateCanceled` ?